nncf/quantization/algo.py

```diff
@@ -912,27 +913,42 @@ def init_range(self):
                 num_init_steps = 0
 
             init_range_config = {'num_init_steps': num_init_steps}
-        if init_range_config:
-            global_init_range_config = dict()
-            global_init_range_config.update(init_range_config)
-            if global_init_range_config.get("type") is None:
-                global_init_range_config["type"] = "mean_min_max"
-
-            num_init_steps = global_init_range_config.get('num_init_steps', 1)
-            if num_init_steps < 0:
-                raise AttributeError('Number of initialization steps must be >= 0')
-            if num_init_steps > 0:
-                try:
-                    range_init_args = self.quantization_config.get_extra_struct(QuantizationRangeInitArgs)
-                except KeyError:
-                    raise ValueError(
-                        'Should run range initialization as specified via config,'
-                        'but the initializing data loader is not provided as an extra struct. '
-                        'Refer to `NNCFConfig.register_extra_structs` and the `QuantizationRangeInitArgs` class')
-                data_loader = range_init_args.data_loader
+        if isinstance(init_range_config, dict):
+            global_init_range_config = self.get_correct_range_config(init_range_config)
+            max_num_init_steps = global_init_range_config['num_init_steps']
+        else:
+            max_num_init_steps = 0
+            global_init_range_config = []
+            for sub_init_range_config in init_range_config:
+                global_init_range_config.append(self.get_correct_range_config(sub_init_range_config))
+                max_num_init_steps = max(sub_init_range_config['num_init_steps'], max_num_init_steps)
 
-                self._do_range_init(data_loader, num_init_steps, global_init_range_config,
-                                    range_init_args.device)
+        if max_num_init_steps > 0:
+            try:
+                range_init_args = self.quantization_config.get_extra_struct(QuantizationRangeInitArgs)
+            except KeyError:
+                raise ValueError(
+                    'Should run range initialization as specified via config,'
+                    'but the initializing data loader is not provided as an extra struct. '
+                    'Refer to `NNCFConfig.register_extra_structs` and the `QuantizationRangeInitArgs` class')
+            data_loader = range_init_args.data_loader
+        
+            self._do_range_init(data_loader, max_num_init_steps, global_init_range_config,
+                                range_init_args.device)
+
+    def get_correct_range_config(self, init_range_config: Dict):
```

Comment:
"correct" is too broad - should be more specific with what kind of config this function returns.

Reply:
Done.